apps/gallery/test/marionette/activity_test.js

```diff
@@ -0,0 +1,130 @@
+    });
+    galleryApp = new Gallery(client);
+    activityCallerApp = GalleryActivityCaller.create(client);
+  });
+
+  test('pick image using gallery pick activity', function() {
+    activityCallerApp.launch();
+    // Initiate pick activity by clicking pick image button
+    activityCallerApp.pickImage();
+
+    // Select image from thumbnail list in gallery app
+    galleryApp.selectImage();
+
+    client.waitFor(function(){
+      return galleryApp.editCropCanvas.displayed();
+    });
+
+    galleryApp.cropDoneButton.click();
+    activityCallerApp.switchTo();
+
+    client.waitFor(function(){
+      return activityCallerApp.pickedImageName.displayed();
+    });
+
+    // Compare returned blob name and type with the
+    // image loaded in gallery app
+    var pickedImageName = activityCallerApp.pickedImageName.text();
+    assert.strictEqual(pickedImageName, imageInfo.name);
+
+    var pickedImageType = activityCallerApp.pickedImageType.text();
+    assert.strictEqual(pickedImageType, imageInfo.type);
+
+  });
+
+  test('share image using gallery share activity', function() {
+    galleryApp.launch();
+    galleryApp.tapFirstThumbnail();
+    galleryApp.shareImage('Fake Gallery Activity Caller');
+    activityCallerApp.switchTo();
+
+    client.waitFor(function(){
+      return activityCallerApp.sharedImageName.displayed();
+    });
+
+    // Compare shared blob name and type with the
+    // image loaded in gallery app
+
+    var sharedImageName = activityCallerApp.sharedImageName.text();
+    assert.strictEqual(sharedImageName, imageInfo.name);
+
+    var sharedImageType = activityCallerApp.sharedImageType.text();
+    assert.strictEqual(sharedImageType, imageInfo.type);
+  });
+
+
+  test('open image using gallery open activity', function() {
+    activityCallerApp.launch();
+    // Initiate pick activity by clicking pick image button
+    activityCallerApp.pickImage();
+
+    // Select image from thumbnail list in gallery app
+    galleryApp.selectImage();
+
+    client.waitFor(function(){
+      return galleryApp.editCropCanvas.displayed();
+    });
+
+    galleryApp.cropDoneButton.click();
+    activityCallerApp.switchTo();
+
+    client.waitFor(function(){
+      return activityCallerApp.pickedImageName.displayed();
+    });
+
+    // Tap picked image to view image using gallery open activity
+    activityCallerApp.tapPickedImage();
+    galleryApp.switchTo();
+
+    // Check if the filename displayed in titlebar matches
+    // data sent by the initiating app
+    var title = galleryApp.openActivityImageTitle.text();
+    assert.strictEqual(title, 'firefoxOS.png');
+
+    // Check if save button is displayed
+    assert.ok(galleryApp.openActivitySaveButton.displayed());
+
+    client.waitFor(function(){
+      return galleryApp.openActivityImage.displayed();
+    });
+
+    // Check if passed blob is set as background in displayed image
+    assert.ok(galleryApp.hasBackgroundImageBlobURL());
```

Comment:
Updated comment to 'Check if displayed image src is set with blob url', I tried but it's not easy to access imageblob property of MediaFrame from activity test.  Need to explore on what's the best way to check the size of opened blob